### PR TITLE
fix(components): [el-dropdown] use custom attributes for dropdown items

### LINKED ISSUE #5768: [Feature Request] Bind custom attrs to ElDropdownItem
### Existing Component
Yes

### Component Name
ElDropdownItem

### Description
Sometimes we want to bind some custom attributes to ElDropdownItem. In old versions it is possible, but it seems to be removed in the latest version. It would be great, if the attrs binding can be back again. Thanks.

image

### PR DESCRIPTION
**Pull request by mawi1512**
Fix https://github.com/element-plus/element-plus/issues/5768

Maybe the issue should be marked as Bug instead of Feature Request as it worked in previous versions.

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/dropdown/src/dropdown-item.vue [hunk 1]
```diff
@@ -5,7 +5,7 @@
   >
     <el-roving-focus-item :focusable="!disabled">
       <el-dropdown-item-impl
-        v-bind="$props"
+        v-bind="$attrs"
```
Comment (JeremyWuuuuu) [inline packages/components/dropdown/src/dropdown-item.vue:8]: I thin we need to do:

```html
v-bind="{...$props, ...$attrs}"
```

Allowing the `$props` to fallthrough.

Comment (JeremyWuuuuu) [reply packages/components/dropdown/src/dropdown-item.vue:8]: @mawi1512 Would you mind change the code as I commented? I think your code are just fine.

Comment (mawi1512) [reply packages/components/dropdown/src/dropdown-item.vue]: @JeremyWuuuuu You are right, $props was missing. I tried to use v-bind="{...$props, ...$attrs}", but it caused type errors. I hope the alternative solution is also ok. Let me know what you think.

File: packages/components/dropdown/src/dropdown-item.vue [hunk 2]
```diff
@@ -97,11 +98,17 @@ export default defineComponent({
       }
     )
 
+    // direct usage of v-bind={ ...$props, ...$attrs } causes type errors
+    const propsAndAttrs = computed(() => {
+      return { ...props, ...attrs }
+    })
```
Comment (JeremyWuuuuu) [inline packages/components/dropdown/src/dropdown-item.vue:104]: **Praise:** I actually want it to be this way.

File: packages/components/dropdown/src/dropdown-item.vue (hunk 1 above)
Comment (JeremyWuuuuu) [reply packages/components/dropdown/src/dropdown-item.vue]: Looking good, test has been triggered, if everything was fine then we should be able to merge this.

**Review by jw-foss — APPROVED** [review]
Let's merge it!

**Pull request merged**